libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+            "foo",
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_cosine_distance(self, aerospike: Aerospike, admin_client: Any) -> None:
+        """Test cosine distance."""
+        from aerospike_vector_search import types
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.COSINE,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz"],
+            ids=["1", "2", "3"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        """
+        foo vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0]
+        far vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0]
+        cosine similarity ~= 0.71
+        cosine distance ~= 1 - cosine similarity = 0.29
+        """
+        expected = pytest.approx(0.292, abs=0.002)
+        output = aerospike.similarity_search_with_score(
+            "far", index_name=index_name, k=3
+        )
+
+        _, actual_score = output[2]
+
+        assert actual_score == expected
+
+    def test_dot_product_distance(
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        """Test dot product distance."""
+        from aerospike_vector_search import types
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.DOT_PRODUCT,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz"],
+            ids=["1", "2", "3"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        """
+        foo vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0]
+        far vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0]
+        dot product = 9.0
+        dot product distance = dot product * -1 = -9.0
+        """
+        expected = -9.0
+        output = aerospike.similarity_search_with_score(
+            "far", index_name=index_name, k=3
+        )
+
+        _, actual_score = output[2]
+
+        assert actual_score == expected
```

Comment:
**Security Issue**: The test for dot product distance does not validate the sign of the dot product which is crucial since the dot product can be negative, affecting the distance calculation. <br> **Fix**: Add validation to check the sign of the dot product and ensure it's handled correctly in the distance calculation. <br> **Code Suggestion**: 
 ```
 +        # Validate the sign of the dot product
 +        if dot_product < 0:
 +            raise ValueError("Dot product should not be negative")
 +        expected = -dot_product
 +        output = aerospike.similarity_search_with_score(
 +            "far", index_name=index_name, k=3
 +        )
 ```